undertow/src/main/java/org/wildfly/extension/undertow/ApplicationSecurityDomainDefinition.java

```diff
@@ -573,67 +585,71 @@ private static HttpScope sessionScope(HttpServerExchange exchange, ScopeSessionL
             ServletRequestContext context = exchange.getAttachment(ServletRequestContext.ATTACHMENT_KEY);
 
             return new HttpScope() {
-                private final AtomicReference<HttpSessionImpl> session = new AtomicReference<>(context.getSession());
+                private HttpSession session = context.getOriginalRequest().getSession();
```

Comment:
Should this be getSession(false) ? getSession() is the same as getSession(true).

I ask because create() below does an exists() check and if false does a getSession(true). But it seems like that exists() check would never return false since ^^^ will create a session.